agent/grpc-external/services/peerstream/stream_test.go

```diff
@@ -1252,8 +1252,8 @@ func TestStreamResources_Server_DisconnectsOnHeartbeatTimeout(t *testing.T) {
 	})
 
 	testutil.RunStep(t, "stream is disconnected due to heartbeat timeout", func(t *testing.T) {
-		disconnectTime := ptr(it.FutureNow(1))
```

Comment:
incrementalTime `it` was being modified in a goroutine so there was a data race, leading `disconnectTime` to be incorrect. I moved the call to be local to each retry loop so that it would fetch the latest "now" value instead of predicting it would be 1 second in the future.